### DISCUSSION TITLE
Customizable switch color (improve styles api or allow for custom color)

### DISCUSSION THREAD

**Discussion by showduhtung** [mantinedev/mantine#3948] [Feature requests]
Currently the `Switch` component uses the `color` prop to customize the component color. But this is limited to the color options that the default/custom theme provides. I tried using all the styles api to set the `backgroundColor` to my preferred color and did make sure to try using the `"&:checked"` pseudo-class, but to no avail. 

Suggestions:
1. Allow the Switch color prop to accept hex code
2. Improve the styles api for the Switch component to access and customize the track `backgroundColor` when checked.

**Comment by yverby** [comment]
Regarding Styles API, you need to style  `track` selector
See [example](https://codesandbox.io/s/kind-heisenberg-xgxzwx?file=/src/App.tsx)

**Reply by showduhtung** [reply]
Ahh I see it now thanks @yverby! Pretty difficult to decipher 1) what styles ref selector I should be looking for and 2) when I should be using it. https://mantine.dev/styles/create-styles/#composition-and-nested-selectors the docs does seem to show how its used, but I don't feel like I was able to really understand _when_ to use it.
